File: 프로그래머스/lv2/12953. N개의 최소공배수/N개의 최소공배수.js
function solution(arr) {
    let answer = 0;
    let target;
    
    const initValue = Math.max(...arr);
    let mulValue = 1;
    while (true) {
        let flag = true;
        target = initValue * mulValue;
        for (let i = 0; i < arr.length; i++) {
            if (target % arr[i] !== 0) {
                flag = false;
                mulValue++;
                break;
            }
        }
        if (flag) break;
    }
    
    return target;
}

File: 프로그래머스/lv2/12953. N개의 최소공배수/N개의 최소공배수.py
def solution(arr):
    answer = 0
    
    maxValue = max(arr)
    target = 1;
    
    while True:
        checkValue = maxValue * target
        flag = False
        for i in range(len(arr)):
            if checkValue % arr[i] != 0:
                flag = True
                target += 1
                break
        if not flag:
            return checkValue
    
